src/SIM/SIMoutput.C

```diff
@@ -1919,6 +1928,50 @@ bool SIMoutput::savePoints (const Vector& psol, double time, int step) const
 }
 
 
+bool SIMoutput::saveResults (const Vector& psol, double time, int step) const
+{
+  if (idxGrid < 0 || step < 1 || psol.empty())
+    return true;
+  else if (idxGrid > static_cast<int>(myPoints.size()))
+    return false;
+
+  const std::string& filName = myPoints[idxGrid].first;
+  const ResPointVec& gridPts = myPoints[idxGrid].second;
+  if (filName.empty() || gridPts.empty())
+    return false;
+
+  size_t idot = filName.find_last_of('.');
+  std::string fileName = filName.substr(0,idot);
+  std::string fileExt  = filName.substr(idot);
+
+  myProblem->initResultPoints(time);
+
+  // Evaluate the solution variables in all grid points
+  IntVec  points;
+  Vec3Vec Xp;
+  Matrix  sol, sol2;
+  for (const ASMbase* pch : myModel)
+    if (!this->evalResults(psol,gridPts,pch,points,Xp,sol,sol2))
+      return false;
+
+  if (!sol.augmentRows(sol2))
+    return false;
+
+  // Write one file for each result component
+  for (size_t r = 1; r <= sol.rows(); r++)
+  {
+    std::string fName = fileName + "_" + std::to_string(r) + fileExt;
+    std::ofstream fs(fName.c_str(), step == 1 ? std::ios::out : std::ios::app);
```

Comment:
ofstream has a string constructor since c++11.

Reply:
Changed to string constructor here and elsewhere in this file